libr/util/rbtree.c

```diff
@@ -36,23 +36,28 @@ static inline RBNode *zig_zag(RBNode *x, int dir, RBNodeSum sum) {
 	return z;
 }
 
-static inline RBIter bound_iter(RBNode *x, void *data, RBComparator cmp, bool upper, bool backward, void *user) {
+static inline RBIter bound_iter(RBNode *x, void *data, RBComparator cmp, bool upper, void *user) {
 	RBIter it;
 	it.len = 0;
 	while (x) {
 		int d = cmp (data, x, user);
-		if (upper ? d < 0 : d <= 0) {
-			if (!backward) {
+		if (d == 0) {
+			it.path[it.len++] = x;
+			return it;
+		}
+		if (d <= 0) {
```

Comment:
just `<`

Reply:
also, use a `else if`, it makes more sense here

Reply:
The first if is a bit of a special case and an early return so I prefer not to have an else here.